### PR TITLE
Plane: Add options to USE_REVERSE_THRUST to cover all flight modes

### PR DESCRIPTION
**Pull request by Hwurzburg**
to address issues in #17040

### CODE REVIEW

**Comment by shellixyz** [comment]
Why not I guess but I think reverse thrust should be enabled by default in manual modes like it was before

**Comment by Hwurzburg** [comment]
> Why not I guess but I think reverse thrust should be enabled by default in manual modes like it was before

but it is not and has not been enabled in any non autothrottle mode except manual..that has been its established behavior since it was added in 11/2018

**Comment by Hwurzburg** [comment]
sorry, I meant when NONE was selected

File: ArduPlane/defines.h [hunk 1]
```diff
@@ -145,6 +145,10 @@ enum {
     USE_REVERSE_THRUST_FBWB                     = (1<<9),
     USE_REVERSE_THRUST_GUIDED                   = (1<<10),
     USE_REVERSE_THRUST_AUTO_LANDING_PATTERN     = (1<<11),
+    USE_REVERSE_THRUST_FBWA                   = (1<<12),
+    USE_REVERSE_THRUST_ACRO                   = (1<<13),
+    USE_REVERSE_THRUST_STABILIZE                   = (1<<14),
+    USE_REVERSE_THRUST_THERMAL            =(1<<15),
```
Comment (tridge) [inline ArduPlane/defines.h:151]: need to also widen use_reverse_thrust to 32 bits, as its signed. We have an easy function now for converting widened parameters

**Review by tridge — CHANGES_REQUESTED** [review]

File: ArduPlane/defines.h (hunk 1 above)
Comment (Hwurzburg) [reply ArduPlane/defines.h:151]: @tridge || @peterbarker  will need a pointer to a pattern for what you want me to do to change it to 32 wide... thanks

Comment (Hwurzburg) [reply ArduPlane/defines.h]: fixed..

**Review by tridge — APPROVED** [review]

**Pull request merged**